tests/undo-redo.tests.js

```diff
@@ -588,3 +588,59 @@ export const testBehaviorOfIgnoreremotemapchangesProperty = tc => {
   t.assert(map1.get('x') === 2)
   t.assert(map2.get('x') === 2)
 }
+
+/**
+ * Ensure that we never stop capturing due to a timeout when the timeout is infinite
+ *
+ * @see https://github.com/yjs/yjs/issues/431
+ * @param {t.TestCase} tc
+ */
+ export const testInfiniteCaptureTimeoutNeverCapturesAutomatically = tc => {
+  const { array0 } = init(tc, { users: 3 })
+  const undoManager = new Y.UndoManager(array0, { captureTimeout: Number.POSITIVE_INFINITY })
+
+  array0.push([1, 2, 3]);  
```

Comment:
Would be nice in the future if the timeout logic were decoupled from the undo manager itself so we could test the behaviour in isolation, without any extra logic related to the shared types getting in the way.